source/common/stats/thread_local_store.h

```diff
@@ -67,8 +67,8 @@ class ThreadLocalStoreImpl : public StoreRoot {
 
   // Stats::StoreRoot
   void addSink(Sink& sink) override { timer_sinks_.push_back(sink); }
-  void setTagExtractors(const std::vector<TagExtractorPtr>& tag_extractors) override {
-    tag_extractors_ = &tag_extractors;
+  void setTagProducer(TagProducerPtr&& tag_producer) override {
+    tag_producer_ = std::move(tag_producer);
```

Comment:
If this is generally meant to be called once, I would add an `ASSERT(tag_producer_ == nullptr);` here.

Reply:
We expects this to be called one. But `tag_producer_` will never be `nullptr` due to this construcor https://github.com/envoyproxy/envoy/pull/2357/files#diff-4112b28a18267da2a53245cfbacaeb94R16.

Reply:
OK no problem just skip it.